src/Aspire.Hosting.AppHost/build/Aspire.Hosting.AppHost.targets

```diff
@@ -255,10 +255,10 @@ namespace Projects%3B
     <MakeDir Directories="$(AspireManifestPublishOutputPath)" />
   </Target>
 
-  <Target Name="__ErrorOnMininumWorkloadVersionMissing" BeforeTargets="PrepareForBuild"
-    Condition="'$(IsAspireHost)' == 'true' and ('$(AspireHostingSDKVersion)' == '' or $([MSBuild]::VersionLessThan('$(AspireHostingSDKVersion)', '8.2.0')))">
-    <Error Code="ASPIRE005"
-      Text="$(MSBuildProjectName) project requires a newer version of the .NET Aspire Workload to work correctly. Please run `dotnet workload update`." />
+  <Target Name="__ErrorOnMininumSdkVersionMissing" BeforeTargets="PrepareForBuild"
+  Condition="'$(IsAspireHost)' == 'true' and ('$(AspireHostingSDKVersion)' == '' or $([MSBuild]::VersionLessThan('$(AspireHostingSDKVersion)', '9.0.0')))">
```

Comment:
@DamianEdwards we should think about whether or not we want to do this. The question is whether or not we want to support the following scenario:

1. User has an existing Aspire 8.x project, and has the workload installed (which is required for them to build the project.)
2. After we release Aspire 9.0, they notice there is a new version of the AppHost package, so they bump their reference to 9.0 without adding a line in their project to use the new MSBuild SDK.

We have two options here:

1. We could just let this succeed, as they would already have the workload installed which practically has the same targets from the SDK so we could let them build. The main pro here is reducing friction for existing customers while upgrading. The main con I see is that we are allowing tearing, and also that this project will succeed or not depending on the machine it is running in.
2. We force them to also set the MSBuild SDK line in their project in order to be able to build. That is the way things are with these changes, but we could change this. The pro here is we don't allow tear reducing our testing matrix, and also that we have consistency across machines as the behavior of the build would be the same whether the 8.x workload is installed or not.

Thoughts?

Reply:
I think we force them to add the MSBuild SDK. To get into this situation they have to change their project file anyway (upgrade to new package version) so it seems best to force them to be fully aligned with the new idiom rather than some odd combination we don't test.

Reply:
> To get into this situation they have to change their project file anyway (upgrade to new package version)

True but only if you assume people update their packages manually, which I'm not so sure that is the most common case. For people doing it via Nuget package manager in VS or some other tool, we would now be forcing them to hand edit the project (as I don't believe there is any UI for doing it)

Reply:
The point was it requires a project modification, so I think it's the right time to force this change.

Reply:
No more forcing users to break, we learned our lesson in 8.2

Reply:
That was different, in that case users were ending up in a broken state without them doing any changes on their side (so we broke them by the only fact of shipping new packages) This one, as @DamianEdwards pointed out, requires a user action to be on this state.

Reply:
Right, this is the key difference. Last time, we broke folks after updating a globally installed asset (the workload). This time, we're saying, if you update your package ref to 9.0, you should update your project file to match what we expect it to look like, i.e. removing the property. We could make it less severe than an error too if that makes us feel better too.

Reply:
I am anti this change. Worst case, there will be turds left in old project files. That seems like a problem that will fix itself as new templates become the default.

Reply:
For full transparency, so we're super clear, the move to 9.0 will require the following changesin the AppHost project:
- Updating the package reference for `Aspire.Hosting.AppHost` to version `9.0`
- Adding the SDK to the project file via an `<Sdk>` item

After doing that, this specific warning would be saying, hey you don't need that `<IsAspireHost>` property. If we think that's a stretch too far and we're OK with that unused property hanging around in some project files, then I'm fine to remove this (but let's do it in a separate PR so we can get this one in!!).

Reply:
Now I've reread the original comment I realize I misunderstood this one. This is about the `<Sdk>` entry itself, for folks who already have the old workload installed. I **really** don't think we should rely on the 8.x workload targets for 9.0 projects, but rather we should force people to move to specifying the SDK specifically in their project files, so that their projects **no longer require the workload**. That is the key change we're trying to make here so we should enforce it IMO. The `<IsAspireHost>` property is a true leftover that has no functional impact so not instructing them to remove that has no impact.

Reply:
Exactly, that describes the state. IsAspireHost will remain and won't hurt (and will help tools like azd as well as our internal conditions in the repo). The question was whether or not we should allow people with existing 8.x projects and workload installed to just bump their AppHost package version and have them be able to build, which we totally could but it does complicate our compat matrix as well as it is prone to break in places where workload isn't installed. Personally, I agree with @DamianEdwards and think we should go ahead with this change.